crates/store/re_dataframe/src/query.rs

```diff
@@ -816,7 +858,9 @@ impl<E: StorageEngineLike> QueryHandle<E> {
             Retrofilled(UnitChunkShared),
         }
 
-        let state = self.init();
+        // That's a synchronous lock, so make sure we barely lock it.
+        let state = self.init_(store, cache);
```

Comment:
This looks suspicious to me -- the body of that function is large and I wouldn't expect us to execute that code unless we aren't able to access `self.state`.

Why isn't this:
```
self.state.get_or_init(|| self.init_(store, cache))
```

Reply:
Uh-oh, nice catch. I meant to add a non-blocking optimistic check that comes first.

But now that I think about it, this is in `QueryHandle`, so realistically nothing will ever contend on this, we can probably just do it all in the OnceLock yeah.